htdocs/takepos/genimg/qr.php

```diff
@@ -29,5 +29,8 @@
 
 $key = GETPOST('key');
 
+$urlwithouturlroot = preg_replace('/'.preg_quote(DOL_URL_ROOT, '/').'$/i', '', trim($dolibarr_main_url_root));
+$urlwithroot = $urlwithouturlroot.DOL_URL_ROOT; // This is to use external domain name found into config file
+
 $module = new modTcpdfbarcode($db);
-$result = $module->buildBarCode("http://www.takepos.com", 'QRCODE', 'Y');
+$result = $module->buildBarCode($urlwithroot."/takepos/auto_order.php?key=".dol_encode($key), 'QRCODE', 'Y');
```

Comment:
Do you mean $urlwithroot."/takepos/public/auto_order.php    ?

Reply:
You are right @eldy 
I had this fix and many other improvements for the next PULL REQUEST